### ISSUE TITLE
correlation_2op_2t with es solver fails when first operator makes density matrix vanish

### ISSUE THREAD

**Issue by yanayyar** [qutip/qutip#500]
Minimal code example:
```
H = Qobj([[-1,0],[0,1]])
a = Qobj([[0,1],[0,0]])
psi0 = Qobj([[1],[0]])
correlation_2op_2t(H,psi0,[0],[0,1],[],a,a,solver='es')
```
(whereas correlation_2op_2t(H,psi0,[0],[0,1],[],a,a.dag(),solver='es') works fine)

From what I can tell, in _correlation_es_2t, solES_tau becomes a 0-term ESERIES object, and then fails to calculate expectation values at various times.

**Comment by kafischer** [comment]
It appeared there was a floating point error in ode2es that would result in an empty amplitude string for small enough amplitudes. In PR #509 this was fixed by forcing the amplitude to zero when the numerical error occurs.

**Comment by nonhermitian** [comment]
I am assuming that this was closed in Pull #509, so closing.
